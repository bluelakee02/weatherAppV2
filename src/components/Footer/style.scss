.footer {
  grid-area: footer;
  background-color: $blue-01;
  display: flex;
  justify-content: space-around;
  align-items: center;

  & h2 {
    background-color: $grey-02;
    color: $grey-01;
  }

}

.footerIcon {
  color: #fff;
  border: 1px solid #fff;
  border-radius: 4px;
  font-size: 2rem;
  padding: .5rem;
  cursor: pointer;
  outline: none;
  margin: 0 10px;

  &.active {
    border: 1px solid $grey-01;
    color: $grey-01;
  }
}

@media screen and (max-width: $breakpoint-tablet) {
  .footer {
    flex-direction: column;
  }
}

