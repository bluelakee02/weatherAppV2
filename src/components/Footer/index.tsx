import React from 'react';

import style from './style.scss';

const Footer: React.FC = () => (
    <footer className={style.footer}>
        <h2>blueLake 2020</h2>
        <div>
            Powered by{' '}
            <a href="https://www.weatherapi.com/" title="Free Weather API">
                WeatherAPI.com
            </a>
        </div>
    </footer>
);

export default Footer;
