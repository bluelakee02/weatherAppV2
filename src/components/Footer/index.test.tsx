import * as React from 'react';
import {shallow, render, mount} from 'enzyme';

import Footer from './index';

describe('Footer test', () => {
  it('Matches the snapshot', () => {
    expect(shallow(<Footer/>)).toMatchSnapshot();
  });

  it('Has proper class', () => {
    const wrapper = mount(<Footer/>);
    expect(wrapper.find(`.footer`)).toHaveLength(1);
  });

  it('Has proper nodes', () => {
    expect(mount(<Footer/>).find('h1').length).toBe(0);
    expect(mount(<Footer/>).find('h2').length).toBe(1);
  });

  it('Renders static HTML', () => {
    expect(render(<Footer/>).text()).toEqual('blueLake 2020thx to MetaWeather');
  });
});
