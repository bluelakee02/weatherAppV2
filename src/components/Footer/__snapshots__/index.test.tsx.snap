// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer test Matches the snapshot 1`] = `
<footer
  className="footer"
>
  <h2>
    blueLake 2020
  </h2>
  <div>
    Powered by
     
    <a
      href="https://www.weatherapi.com/"
      title="Free Weather API"
    >
      WeatherAPI.com
    </a>
  </div>
</footer>
`;
